spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+  describe "GET /index" do
+    let(:valid_attributes) {{ name: 'Alice', photo: 'http://', bio: 'Hello', posts_counter: 0}}
+    it "render a successful response" do
+      user = User.create! valid_attributes
+      get :index
+      expect(response).to be_successful
+    end
+
+    describe "GET #show" do
+      let(:user) { User.create!(name: "John Doe", posts_counter: 0) }
+
+      it "renders a successful response" do
+        get :show, params: { id: user.id }
+        expect(response).to be_successful
+      end
+
+      it "renders the show template" do
+        get :show, params: { id: user.id }
+        expect(response).to render_template('users/profile')
+      end
+  
```

Comment:
- [Optional] Dear @hienphan0111, Your tests are great, I like you have tested for almost all the view templates that ensure the correct view template is rendered, but I note you missed a `render_template('users')` test for the user index page.  Consider adding it for a complete test.